var/spack/repos/builtin/packages/hdf-eos2/package.py

```diff
@@ -63,32 +76,48 @@ class HdfEos2(AutotoolsPackage):
     depends_on("jpeg")
     depends_on("szip", when="^hdf +szip")
 
+    # Fix some problematic logic in stock configure script
+    # test succeeds, but then script aborts due to env variable not being set
+    patch("hdf-eos2.configure.patch", when="@2:3.0")
+
     # The standard Makefile.am, etc. add a --single_module flag to LDFLAGS
     # to pass to the linker.
     # That appears to be only recognized by the Darwin linker, remove it
     # if we are not running on darwin/
     if sys.platform != "darwin":
-        patch("hdf-eos2.nondarwin-no-single_module.patch")
+        patch("hdf-eos2.nondarwin-no-single_module.patch", when="@2")
 
     def url_for_version(self, version):
         vrec = [x for x in self.version_list if x["version"] == version.dotted.string]
         if vrec:
             fname = vrec[0]["basename"].format(version.dotted, version.underscored)
             sha256 = vrec[0]["sha256"]
-            myurl = self.url.format(sha256, fname)
+            can_auto_download = vrec[0].get("can_auto_download", False)
```

Comment:
The package management for hdf-eos2 is obviously "unique" and creates its challenges.

I am wondering if it is wise to tie the `can_auto_download` logic to the version. These things could change I suppose? Would it be better to set the url in `url_for_version` based on the version requested, and also set `manual_download` based on the version? There wouldn't be any need for `can_auto_download` at all.